src/main/java/io/codeka/gaia/repository/TerraformCLIRepository.java

```diff
@@ -0,0 +1,112 @@
+package io.codeka.gaia.repository;
+
+import org.apache.commons.lang.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Repository;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+@Repository
+public class TerraformCLIRepository {
+
+    private static final Pattern TERRAFORM_VERSION_PATTERN = Pattern.compile("terraform_([\\d]+.[\\d]+.[\\d]+)");
+    private static final Pattern SEMVER_PATTERN = Pattern.compile("([\\d]*).([\\d]*).([\\d]*)");
+
+    private String terraformReleasesUrl;
+    private String terraformReleasesVersionMin;
+    private RestTemplate restTemplate;
+
+    private class CLIVersion {
+        private Integer major;
```

Comment:
Why using objects instead of primitive types?
Is there a specific reason?

```suggestion
        private int major;
```

Reply:
old habit.
I put `short`